File: Problemset/0020-Valid Parentheses/Remove Valid Parentheses by Stack.py
class Solution:
    def isValid(self, s: str) -> bool:
        if len(s) % 2 != 0:
            return False
        brackets = {')': '(', '}': '{', ']': '['}
        stack = []
        for c in s:
            if c in brackets and stack and stack[-1] == brackets[c]:
                stack.pop()
            else:
                stack.append(c)
        return not stack

File: Problemset/0921-Minimum Add to Make Parentheses Valid/Remove Valid Parentheses by Stack.rs
impl Solution {
    pub fn min_add_to_make_valid(s: String) -> i32 {
        let mut stack: Vec<char> = Vec::new();
        for ch in s.chars() {
            if ch == ')' && stack.ends_with(&['(']) {
                stack.pop();
            } else {
                stack.push(ch);
            }
        }
        stack.len() as i32
    }
}
